src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListService.java

```diff
@@ -98,4 +99,12 @@ void removeGermplasmEntriesFromList(Integer germplasmListId,
 
 	Map<Integer, Map<Integer, String>> getObservationValuesByListAndVariableIds(Integer listId, Set<Integer> variableIds);
 
+	/**
+	 * Inserts a list of multiple {@code GermplasmListData} objects into the database.
+	 *
+	 * @param data - A list of {@code GermplasmListData} objects to be persisted to the database. {@code GermplasmListData}
+	 *             objects must be valid.
+	 * @return Returns the ids of the {@code GermplasmListData} records inserted in the database.
+	 */
+	List<GermplasmListData> addGermplasmListData(List<GermplasmListData> data);
```

Comment:
@cuenyad please do not expose this only to be used in a test. Call DAO directly using a simplified version of the implementation

Reply:
done